1298 acl access control list augmentation

afa7789 opened the pull request:
Adding new features to ACL to make easier to use.

- list of ACL data, to debug.
- README updated with new info.
- added new functions to create a history of transactions in acl.
- at normal acl operations added feature to add transactions to this history table.
- option at acl mode CLI to add new config variable value to the config table.
- added at startup a log of last X acl transactions changes ( 10 being the default ).

afa7789 committed:
feat: adding list command to check ACL status on acl folder.

also function can be reused by other parts of the code.

afa7789 committed:
feat: adding ListComponent calls in cdk-erigon

added call to the function for it to log in parts of the code, this will be reviewed.

afa7789 committed:
docs: adding readme updates on acl, making it more clear on how to use.

afa7789 committed:
feat: acl transaction history buildup

creating functions to interact with table as binary, and other helper functions that we may need.

afa7789 committed:
feat: build up transaction inserter, second part.

afa7789 committed:
feat: enhance ACL listing with policy transaction logging and refactor logging imports

changing log structure, cause it's failing.

changed the function at txpool so I can get the string to put in the log at the CLI level since it doesn't log too.

afa7789 committed:
chore: changing call to log it.

afa7789 committed:
feat: update logging for policy transactions and fix a transaction insertion error

afa7789 committed:
feat: add update operation to policy transactions and log updates in UpdatePolicies

afa7789 committed:
feat: insert policy refactor and fix update error.

fixing update method which was exiting on panic, also improved the function of insert to be able to insert an array improving code quality

afa7789 committed:
feat: adding log count setup to the mode acl function ( which basically is the config updater now)

afa7789 committed:
chore: changing config value on key to string, for easier reading when debugging// looking for it.

not an issue for usage, only a improvement.

afa7789 commented:
<img width="622" alt="image" src="https://github.com/user-attachments/assets/2a66eab4-54c1-4a93-81fb-e405c6036060">
<img width="1710" alt="image" src="https://github.com/user-attachments/assets/1a8004e7-0e2c-4a5f-93b2-4f4a67ea70c6">

afa7789 committed:
docs: update README to include optional log_count parameter for acl mode command

hexoscott commented on cmd/acl/main.go:
```diff
@@ -7,24 +7,23 @@ import (
 
 func main() {
-	logging.LogVerbosityFlag.Value = log.LvlError.String()
-	logging.LogConsoleVerbosityFlag.Value = log.LvlError.String()
```
could removing these lines be the reason log/v3 isn't working as expected?

hexoscott reviewed (changes requested):
Changes look good, but we need to address the logging thing before we can merge

afa7789 committed:
chore: adding logging level part back.

hexoscott commented:
Hi @afa7789 - are we considering this good to review again?  Did adding the log level stuff back in help with the v3 logging library problem you were seeing?

afa7789 committed:
fix: we had an issue in the test when running in parallel

it could either remove or add the policy at the same time someone else was checking for another scenario

to fix I added the adding flow, to updating/changing tx before making the update with the same asserting checks for correct adding.
